platform/platform-1_20_6/src/main/java/tc/oc/pgm/platform/v1_20_6/material/ModernColorUtils.java

```diff
@@ -0,0 +1,142 @@
+package tc.oc.pgm.platform.v1_20_6.material;
+
+import static tc.oc.pgm.util.platform.Supports.Variant.PAPER;
+
+import java.util.EnumMap;
+import java.util.HashMap;
+import java.util.Map;
+import net.kyori.adventure.text.Component;
+import org.bukkit.Bukkit;
+import org.bukkit.ChatColor;
+import org.bukkit.DyeColor;
+import org.bukkit.Location;
+import org.bukkit.Material;
+import org.bukkit.block.Banner;
+import org.bukkit.block.Block;
+import org.bukkit.block.BlockFace;
+import org.bukkit.block.BlockState;
+import org.bukkit.block.data.Rotatable;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.BannerMeta;
+import org.bukkit.scoreboard.Team;
+import tc.oc.pgm.util.block.BlockFaces;
+import tc.oc.pgm.util.material.ColorUtils;
+import tc.oc.pgm.util.platform.Supports;
+
+@Supports(value = PAPER, minVersion = "1.20.6")
+@SuppressWarnings("deprecation")
+public class ModernColorUtils implements ColorUtils {
+
+  private static final Map<Material, Map<DyeColor, Material>> COLORABLE_MATERIALS = new HashMap<>();
+
+  static {
+    registerColorable("WOOL", Material.LEGACY_WOOL);
+    registerColorable("DYE", Material.LEGACY_INK_SACK);
+    registerColorable("CARPET", Material.LEGACY_CARPET);
+    registerColorable("TERRACOTTA", Material.LEGACY_STAINED_CLAY);
+    registerColorable("GLAZED_TERRACOTTA");
+    registerColorable("STAINED_GLASS", Material.LEGACY_STAINED_GLASS);
+    registerColorable("STAINED_GLASS_PANE", Material.LEGACY_STAINED_GLASS_PANE);
+
+    registerColorable("BANNER", Material.LEGACY_BANNER, Material.LEGACY_STANDING_BANNER);
+    registerColorable("WALL_BANNER", Material.LEGACY_WALL_BANNER);
+    // TODO: Any others?
```

Comment:
We could consider some sort of automatic detection based off of colors in the material names

Reply:
For the time being i've added all i could find, can elaborate on it further later